From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Fri, 13 Jan 2017 21:16:30 +0100
Subject: [PATCH] patch-packages: cirros-testvm: Use Armband version

Armband repackages cirros-testvm to add AArch64 binaries, and hence
rebuilds x86 binaries too, to align package versions.

Account for Armband's cirros_testvm version by using its specific
version string instead of the old MOS one.

Now Armband also patches test-vm with an updated x86 image in the
upstream repo, so we can disable patch-packages for cirros-testvm
during ISO build.

JIRA: ARMBAND-200

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 build/Makefile                              | 4 ++--
 build/patch-packages/Makefile               | 3 ++-
 build/patch-packages/cirros_testvm/Makefile | 2 +-
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/build/Makefile b/build/Makefile
index 4985bdf..db8661b 100644
--- a/build/Makefile
+++ b/build/Makefile
@@ -175,7 +175,7 @@ patch-packages:
 .PHONY: clean $(SUBCLEAN)
 clean:  $(SUBCLEAN)
 	$(MAKE) -C f_repos -f Makefile clean
-	$(MAKE) -C patch-packages -f Makefile clean
+	#$(MAKE) -C patch-packages -f Makefile clean
 	@rm -f *.iso
 	@rm -Rf release
 	@rm -Rf newiso
@@ -211,7 +211,7 @@ setup-env:
 # Todo: Make things smarter - we shouldn't need to clean everything
 # betwen make invocations.
 .PHONY: iso
-iso:	setup-env $(ISOCACHE) $(SUBDIRS) patch-packages
+iso:	setup-env $(ISOCACHE) $(SUBDIRS)
 	$(REPOINFO) . > gitinfo_main.txt
 	install/install.sh iso $(ISOCACHE) $(NEWISO) $(PRODNO) $(REVSTATE)
 	@printf "\n\nProduct ISO is $(NEWISO)\n\n"
diff --git a/build/patch-packages/Makefile b/build/patch-packages/Makefile
index aaac038..bb450ad 100644
--- a/build/patch-packages/Makefile
+++ b/build/patch-packages/Makefile
@@ -8,7 +8,8 @@
 # http://www.apache.org/licenses/LICENSE-2.0
 ##############################################################################

-SUBDIRS := cirros_testvm
+# NOTE(armband): Fuel@OPNFV patches cirros_testvm @ ISO build, we don't.
+SUBDIRS :=
 SUBCLEAN = $(addsuffix .clean,$(SUBDIRS))

 .PHONY: $(SUBDIRS) $(SUBCLEAN) clean
diff --git a/build/patch-packages/cirros_testvm/Makefile b/build/patch-packages/cirros_testvm/Makefile
index b6a56d4..b0163d0 100644
--- a/build/patch-packages/cirros_testvm/Makefile
+++ b/build/patch-packages/cirros_testvm/Makefile
@@ -23,7 +23,7 @@ clean:

 .PHONY: release
 release:
-	../tools/deb_unpack c/cirros-testvm/cirros-testvm_0.3.4-2~u16.04+mos5_amd64.deb $(ORIGISO)
+	../tools/deb_unpack c/cirros-testvm/cirros-testvm_0.3.4-2+amos2~u16.04+mos5_amd64.deb $(ORIGISO)
 	@rm -rf package/usr/share/cirros-testvm/*
 	wget -O package/usr/share/cirros-testvm/cirros-x86_64-disk.img http://download.cirros-cloud.net/daily/20160722/cirros-d160722-x86_64-disk.img
 	../tools/deb_pack $(REVSTATE)
